src/Sylius/Bundle/AttributeBundle/AttributeType/TextAttributeType.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AttributeBundle\AttributeType;
+
+use Sylius\Component\Attribute\AttributeType\AttributeTypeInterface;
+use Sylius\Component\Attribute\Model\AttributeValueInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class TextAttributeType implements AttributeTypeInterface
+{
+    const TYPE = 'text';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStorageType()
+    {
+        return AttributeValueInterface::STORAGE_TEXT;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType()
+    {
+        return static::TYPE;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate(AttributeValueInterface $attributeValue, ExecutionContextInterface $context)
```

Comment:
I'd actually just pass ``array $configuration`` as 3rd argument. This way developer does not need to do:

``$attributeValue->getAttribute()->getConfguration()``

Just like in form types etc.